pydantic/validators.py

```diff
@@ -404,6 +404,21 @@ def arbitrary_type_validator(v: Any) -> T:
     return arbitrary_type_validator
 
 
+def make_class_validator(type_: Type[T]) -> Callable[[Any], Type[T]]:
+    def class_validator(v: Any) -> Type[T]:
+        if issubclass(v, type_):
```

Comment:
What happens here if you pass something that isn't a type, eg. `1`? You might need to use `lenient_issubclass`?

Reply:
good catch :)
added a test and fixed it.